check-config-win.py

```diff
@@ -1,64 +1,123 @@
 import os
+import subprocess
 import winreg
 
-key_path = "Software\\Mozilla\\NativeMessagingHosts\\ff2mpv"
-# Assuming current user overrides local machine.
-key_roots = ["HKEY_CURRENT_USER", "HKEY_LOCAL_MACHINE"]
+# Command-Line
+parser = argparse.ArgumentParser(description="Helper for ff2mpv on windows.")
+group = parser.add_mutually_exclusive_group(required=True)
+group.add_argument(
+    "-c",
+    "--check",
+    action="store_true",
+    help="only checks the installation, no modification",
+)
+group.add_argument(
+    "-i",
+    "--install",
+    action="store_true",
+    help="installs ff2mpv registry key or updates the path value",
+)
+group.add_argument(
+    "-u",
+    "--uninstall",
+    action="store_true",
+    help="removes ff2mpv registry key and all it's values",
+)
+args = parser.parse_args()
 
+WDIR = os.path.dirname(__file__)
+FF2MPV_JSON = fr"{WDIR}\ff2mpv-windows.json"
+FF2MPV_KEY = r"Software\Mozilla\NativeMessagingHosts\ff2mpv"
+# Assuming current user overrides local machine.
+HKEYS = {
+    "HKEY_CURRENT_USER": winreg.HKEY_CURRENT_USER,
+    "HKEY_LOCAL_MACHINE": winreg.HKEY_LOCAL_MACHINE,
+}
+error = False
 found_key = False
-
-for root in key_roots:
-    key = winreg.OpenKey(getattr(winreg, root), key_path)
+print("- Checking Registry:")
+for k in HKEYS:
     try:
-        print("Checking:", root, key_path)
-        res = winreg.QueryValueEx(key, "")
+        print(fr"{k}\{FF2MPV_KEY} ... ", end="")
+        key_open = winreg.OpenKey(HKEYS[k], FF2MPV_KEY)
+        hkey_found = HKEYS[k]
+        print("Found.")
     except FileNotFoundError:
-        print("...error finding key")
+        print("Not found.")
+        error = True
         continue
-
+    error = False
     found_key = True
     break
 
 if not found_key:
-    raise ValueError("Could not find a registry entry, aborting.")
+    if args.install:
+        # The intermediate missing key are also created.
+        key_open = winreg.CreateKey(HKEYS["HKEY_CURRENT_USER"], FF2MPV_KEY)
+        print("Key created.")
 
-json_path = res[0]
-print("Path from registry key is:", json_path)
-if not os.path.exists(json_path):
-    raise ValueError("JSON file does not exist:", json_path)
+if not args.uninstall:
+    ff2mpv_value = winreg.QueryValue(key_open, "")
+    if args.install:
+        if ff2mpv_value != FF2MPV_JSON:
+            winreg.SetValue(
+                HKEYS["HKEY_CURRENT_USER"], FF2MPV_KEY, winreg.REG_SZ, FF2MPV_JSON
+            )
+            ff2mpv_value = winreg.QueryValue(key_open, "")
+            print("Value set/updated.\nRestart Firefox if it was running.")
+        else:
+            print("Nothing to update.")
 
-try:
-    bat_data = json.load(open(json_path, "r"))
-except json.decoder.JSONDecodeError:
-    raise ValueError("Parsing error. Is {} a JSON file?".format(json_path))
+    else:
+        if ff2mpv_value != "":
+            print("Value of the key is:", ff2mpv_value)
+            try:
+                os.path.exists(ff2mpv_value)
```

Comment:
doesn't os.path.exists return a bool, and not throw an exception?

Reply:
Yes, but with `try:`, i suppose the system just return a `FileNotFoundError:`.
I have change it to a simple `if: else:`, seems cleaner.
It work in both case.